kubernetes_platform/python/kfp/kubernetes/image.py

```diff
@@ -36,11 +36,17 @@ def set_image_pull_secrets(
 
     msg = common.get_existing_kubernetes_config_as_message(task)
 
-    # Assuming secret_names is a list of strings
-    image_pull_secret = [
-        pb.ImagePullSecret(secret_name=secret_name)
-        for secret_name in secret_names
-    ]
+    if is_list_of_strings(secret_names) or is_list_of_parameter_channel(secret_names):
+        image_pull_secret = []
+        for secret_name in secret_names:
+            secret_name_parameter = common.parse_k8s_parameter_input(secret_name, task)
+            image_pull_secret_pb = pb.ImagePullSecret(secret_name_parameter=secret_name_parameter)
+            if isinstance(secret_name, str):
+                image_pull_secret_pb.secret_name = secret_name
+            image_pull_secret.append(image_pull_secret_pb)
+    else:
+        raise ValueError("secret_names must be a list of strings or "
```

Comment:
It seems reasonable to have a mix of strings and parameters. Is there a reason to not allow this?

Reply:
hmm I agree, I can't remember why I went this route, maybe a limitation on backend, I'll double check and update it if there was a good reason lol